src/coreclr/jit/valuenum.cpp

```diff
@@ -4580,20 +4581,94 @@ ValueNum ValueNumStore::EvalMathFuncUnary(var_types typ, NamedIntrinsic gtMathFN
                 case NI_System_Math_Abs:
                     res = fabs(arg0Val);
                     break;
+
+                case NI_System_Math_Acos:
+                    res = acos(arg0Val);
```

Comment:
It would be "nice" if these could call `COMDouble::Acos`, etc. However I think that's difficult due to layering and it wasn't obvious what would be needed to make that happen.